### PR TITLE
Add an option to specify RSA key size to cf-key

### PR DESCRIPTION
**Pull request by amousset**
First draft of a new option to set the size of the key generated by cf-key. Do you think "-b" is ok for this?

### CODE REVIEW

**Comment by jimis** [comment]
Thanks!

How about `-T` or `--key-type`? For now it will only be a number signifying the RSA key size. In the future it could be more complicated like `ecdsa:256` etc. @nickanderson what do you think?

**Comment by nickanderson** [comment]
--key-type seems OK to me. I don't really care for single letter options.

**Comment by amousset** [comment]
Changed option to `key-type`.

File: cf-key/cf-key.c [hunk 1]
```diff
@@ -249,6 +252,10 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
             KEY_PATH = optarg;
             break;
 
+        case 'T':
+            KEY_SIZE = atoi(optarg);
```
Comment (jimis) [inline cf-key/cf-key.c]: You need some error checking here. @olehermanse what is the function you suggest here?

Comment (olehermanse) [reply cf-key/cf-key.c]: `KEY_SIZE = StringToLongExitOnError(optarg);`

**Comment by amousset** [comment]
Also changed `atoi` to `StringToLongExitOnError` in cf-runagent (from where it was initially copied).

**Comment by olehermanse** [comment]
Thanks @amousset. I will run this through our build/test infrastructure now.

**Comment by nickanderson** [comment]
@olehermanse the pr pipeline you referenced notes 33,689 tests 0 failed.

**Pull request merged**